plutus-core/src/Language/PlutusCore/Parser.y

```diff
@@ -190,6 +167,86 @@ getStaticBuiltinName = \case
     "ifThenElse"               -> Just IfThenElse
     _                          -> Nothing
 
+
+-- TODO: somehow allow the type names to be provided from outside
+mkBuiltinType
+    :: DefaultUni <: uni
+    => AlexPosn -> String -> Parse (Type TyName uni AlexPosn)
+mkBuiltinType tyloc tyname = case tyname of
+  "bool"       -> pure $ mkTyBuiltin @Bool       tyloc
+  "bytestring" -> pure $ mkTyBuiltin @ByteString tyloc
+  "char"       -> pure $ mkTyBuiltin @Char       tyloc
+  "integer"    -> pure $ mkTyBuiltin @Integer    tyloc
```

Comment:
Can we do a similar thing to how we parse static built-in names? I.e. enumerate the entire universe of types, pretty-print each of them and look the `tyname` up in this list/`Map`? There has recently been a [post mentioning this trick](https://kowainik.github.io/posts/haskell-mini-patterns#bidirectional-parsing), by the way.

We can precompute that `Map` from `uni` and simply pass it to `mkBuiltinType` instead of recomputing the `Map` every time we enter `mkBuiltinType`.

We'd need a cleverer strategy to support infinite universes, but it should be fine to only support finite ones for now.

Reply:
Or we could have something like ```uni `Everywhere` Parseable```

Reply:
Yes, that's coming, as it says above. This is just a preliminary step to get started on this.